### PR TITLE
Remove wasm solvers

### LINKED ISSUE #498: Remove wasm helpers
Wasm helpers are not being used as the vision they are based on (user defined directives) hasn't materialised. While they were not restricting us, now they start being limiting as they are not generic over curves.

### PR DESCRIPTION
**Pull request by Schaeff**
closes #498

### CODE REVIEW

File: zokrates_core/src/absy/from_ast.rs [hunk 1]
```diff
@@ -789,7 +789,7 @@ mod tests {
     mod types {
         use super::*;
 
-        /// Helper method to generate the ast for `def main(private {ty} a) -> (): return` which we use to check ty
+        /// solver method to generate the ast for `def main(private {ty} a) -> (): return` which we use to check ty
```
Comment (Schaeff) [inline zokrates_core/src/absy/from_ast.rs]:
```suggestion
        /// Helper method to generate the ast for `def main(private {ty} a) -> (): return` which we use to check ty
```

File: zokrates_core/src/solvers/mod.rs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl Signed for Solver {
+    fn get_signature(&self) -> (usize, usize) {
+        match self {
+            Solver::ConditionEq => (1, 2),
+            Solver::Bits => (1, 254),
```
Comment (JacobEberhardt) [inline zokrates_core/src/solvers/mod.rs]: As this depends on the field, this should be the bitwidth of the prime p. Not this PR, but important.

File: zokrates_core/src/solvers/mod.rs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+        let res = match self {
+            Solver::ConditionEq => match inputs[0].is_zero() {
+                true => vec![T::zero(), T::one()],
+                false => vec![T::one(), T::one() / inputs[0].clone()],
+            },
+            Solver::Bits => {
+                let mut num = inputs[0].clone();
+                let mut res = vec![];
+                let bits = 254;
```
Comment (JacobEberhardt) [inline zokrates_core/src/solvers/mod.rs]: make variable.

File: zokrates_core/src/solvers/mod.rs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            let inputs = vec![1];
+            let r = cond_eq
+                .execute(&inputs.iter().map(|&i| FieldPrime::from(i)).collect())
+                .unwrap();
+            let res: Vec<FieldPrime> = vec![1, 1].iter().map(|&i| FieldPrime::from(i)).collect();
+            assert_eq!(r, &res[..]);
+        }
+    }
+
+    #[test]
```
Comment (JacobEberhardt) [inline zokrates_core/src/solvers/mod.rs]: test should depend on `p`s bitwidth.

File: zokrates_core/src/standard.rs [hunk 5]
```diff
@@ -1,6 +1,6 @@
 use crate::flat_absy::{FlatExpression, FlatExpressionList, FlatFunction, FlatStatement};
 use crate::flat_absy::{FlatParameter, FlatVariable};
-use crate::helpers::{DirectiveStatement, Helper, RustHelper};
+use crate::solvers::{FlatDirective, solver, Rustsolver};
```
Comment (JacobEberhardt) [inline zokrates_core/src/standard.rs]: Shouldn't `Rustsolver` be CamelCase, i.e., `RustSolver`?

File: zokrates_core/src/solvers/mod.rs (hunk 2 above)
Comment (Schaeff) [reply zokrates_core/src/solvers/mod.rs]: Yep, handled in #493

File: zokrates_core/src/solvers/mod.rs (hunk 3 above)
Comment (Schaeff) [reply zokrates_core/src/solvers/mod.rs]: #493

File: zokrates_core/src/solvers/mod.rs (hunk 4 above)
Comment (Schaeff) [reply zokrates_core/src/solvers/mod.rs]: #493

File: zokrates_core/src/standard.rs (hunk 5 above)
Comment (Schaeff) [reply zokrates_core/src/standard.rs]: Good catch, fixing this

**Comment by Schaeff** [comment]
@JacobEberhardt can we merge this?

**Pull request merged**